qiita_pet/handlers/rest/study_person.py

```
@@ -15,18 +17,26 @@
 class StudyPersonHandler(RESTHandler):
     @authenticate_oauth
     def get(self, *args, **kwargs):
-        name = self.get_argument('name')
-        affiliation = self.get_argument('affiliation')
-
-        try:
-            p = StudyPerson.from_name_and_affiliation(name, affiliation)
-        except QiitaDBLookupError:
-            self.fail('Person not found', 404)
-            return
-
-        self.write({'address': p.address, 'phone': p.phone, 'email': p.email,
-                    'id': p.id})
-        self.finish()
+        name = self.get_argument('name', None)
+        affiliation = self.get_argument('affiliation', None)
+
+        if name is None and affiliation is None:
+            # Retrieve the list of all the StudyPerson
+            sp = [{'name': p.name, 'affiliation': p.affiliation}
+                  for p in StudyPerson.iter()]
+            self.write(json_encode(sp))
+            self.finish()
+        elif name is not None and affiliation is not None:
+            try:
+                p = StudyPerson.from_name_and_affiliation(name, affiliation)
+            except QiitaDBLookupError:
+                self.fail('Person not found', 404)
+                return
+            self.write({'address': p.address, 'phone': p.phone,
+                        'email': p.email, 'id': p.id})
+            self.finish()
+        else:
+            self.fail('Missing parameter', 400)
```

I think this should be a raise `MissingArgumentError`?